api/api_public_transaction_pool.go

```diff
@@ -556,6 +556,14 @@ func resend(s *PublicTransactionPoolAPI, ctx context.Context, sendArgs NewTxArgs
 // RecoverFromTransaction recovers the sender address from a signed raw transaction.
 // The signature is validated against the sender account's key configuration at the given block number.
 func (s *PublicTransactionPoolAPI) RecoverFromTransaction(ctx context.Context, encodedTx hexutil.Bytes, blockNumber rpc.BlockNumber) (common.Address, error) {
+	if len(encodedTx) == 0 {
+		return common.Address{}, fmt.Errorf("Empty input")
+	}
+	if 0 < encodedTx[0] && encodedTx[0] < 0x8 { // TODO: Add helper to distinguish eth vs klay txtypes
```

Comment:
Geth seems to have only four transaction types. Is the expected range for `encodedTx[0]` supposed to be within the bounds of `0 <= encodedTx[0] <= 3`? What should be the interpretation or handling of values 4, 5, 6, and 7 in this context?

Reply:
4,5,6,7 will be first extended to 0x7804, ..., 0x7807, then should fail with errUndefinedTxType, as stated in https://github.com/klaytn/klaytn/blob/v1.11.1/blockchain/types/tx_internal_data.go#L484. That is the behavior before and after this PR.

I'd like to leave the remaining fixes for (1) defining the eth vs. klay TxType enum boundary and (2) the same problem in klay_sendRawTransaction at later times.

Reply:
Got it thanks.